crates/algokit_utils/src/transactions/sender_results.rs

```diff
@@ -0,0 +1,429 @@
+use algod_client::models::PendingTransactionResponse;
+use algokit_abi::ABIReturn;
+use algokit_transact::{
+    Address, ApplicationCallTransactionFields, AssetConfigTransactionFields, Transaction,
+    TransactionExt,
+};
+
+/// The unified, comprehensive result of sending a single transaction or transaction group.
+///
+/// This struct provides complete information about the transaction(s) that were sent,
+/// including parsed transaction data, confirmations, and group context. It replaces
+/// all the individual result structs with a single, powerful interface.
+#[derive(Debug, Clone)]
+pub struct SendTransactionResult {
+    // Convenience accessors for the primary transaction (last in group)
+    /// The transaction data for the primary transaction
+    pub transaction: Transaction,
+    /// The confirmation for the primary transaction
+    pub confirmation: PendingTransactionResponse,
+    /// The transaction ID of the primary transaction
+    pub tx_id: String,
+
+    // Full context for the entire group
+    /// The group ID for the transaction group
+    pub group_id: String,
+    /// All transaction IDs in the group
+    pub tx_ids: Vec<String>,
+    /// All transactions in the group
+    pub transactions: Vec<Transaction>,
+    /// All confirmations in the group
+    pub confirmations: Vec<PendingTransactionResponse>,
+
+    // ABI support
+    /// ABI return values for application calls (if any)
+    pub abi_returns: Option<Vec<ABIReturn>>,
+}
+
+/// Result of sending an asset creation transaction.
+///
+/// This is a specialized result that includes the asset ID extracted from the confirmation.
+#[derive(Debug, Clone)]
+pub struct SendAssetCreateResult {
+    /// The base transaction result containing all standard information
+    pub base: SendTransactionResult,
+    /// The ID of the newly created asset (extracted from confirmation)
+    pub asset_id: u64,
+}
+
+/// Result of sending an application creation transaction.
+///
+/// This is a specialized result that includes the app ID and address extracted from the confirmation.
+#[derive(Debug, Clone)]
+pub struct SendAppCreateResult {
+    /// The base transaction result containing all standard information
+    pub base: SendTransactionResult,
```

Comment:
We use `common_params` in Composer transactions. I'm not sure if `base` or `common_params` is better, but we should probably keep consistent. Is there a rust standard for this?